multiqc/modules/slamdunk/slamdunk.py

```diff
@@ -0,0 +1,832 @@
+        basic stats table at the top of the report """
+
+
+        headers = OrderedDict()
+        headers['sequenced'] = {
+            'title': 'Sequenced',
+            'description': '# sequenced reads',
+            'shared_key': 'slamdunk_reads',
+            'min': 0,
+            'format': '{:.f}'
+        }
+        headers['mapped'] = {
+            'title': 'Mapped',
+            'description': '# mapped reads',
+            'shared_key': 'slamdunk_reads',
+            'min': 0,
+            'format': '{:.f}'
+        }
+#         headers['deduplicated'] = {
+#             'title': 'Deduplicated',
+#             'description': '# deduplicated reads',
+#             'shared_key': 'slamdunk_reads',
+#             'min': 0,
+#             'format': '{:.f}'
+#         }
+        headers['mqfiltered'] = {
+            'title': 'MQ-Filtered',
+            'description': '# MQ-filtered reads',
+            'shared_key': 'slamdunk_reads',
+            'min': 0,
+            'format': '{:.f}'
+        }
+        headers['idfiltered'] = {
+            'title': 'Identity-Filtered',
+            'description': '# identity-filtered reads',
+            'shared_key': 'slamdunk_reads',
+            'min': 0,
+            'format': '{:.f}'
+        }
+        headers['nmfiltered'] = {
+            'title': 'NM-Filtered',
+            'description': '# NM-filtered reads',
+            'shared_key': 'slamdunk_reads',
+            'min': 0,
+            'format': '{:.f}'
+        }
+        headers['multimapper'] = {
+            'title': 'Multimap-Filtered',
+            'description': '# multimap-filtered reads',
+            'shared_key': 'slamdunk_reads',
+            'min': 0,
+            'format': '{:.f}'
+        }
+        headers['retained'] = {
+            'title': 'Retained',
+            'description': '# retained reads after filtering',
+            'shared_key': 'slamdunk_reads',
+            'min': 0,
+            'format': '{:.f}'
+        }
+        
+        if (extendedSummary) :
+            headers['counted'] = {
+                'title': 'Counted',
+                'description': '# reads counted within 3\'UTRs',
+                'shared_key': 'slamdunk_reads',
+                'min': 0,
+                'format': '{:.f}'
+            }
+        self.general_stats_addcols(self.slamdunk_data, headers)
+        
+    def slamdunkOverallRatesPlot (self):
+        """ Generate the overall rates plot """
+
+        pconfig = {
+            'id': 'overallratesplot',
+            'title': 'Overall conversion rates in reads',
+            'cpswitch' : False,
+            'cpswitch_c_active': False,
+            'stacking' : 'normal',
+            'data_labels': [
+                "Plus Strand +",
+                "Minus Strand -", 
+            ]
+        }
+        
+        cats = [OrderedDict(), OrderedDict()]
+        
+        cats[0]['T>C'] = {
+            'name': 'T>C',
```

Comment:
Nope, I need it. If I remove it, this happens:

[ERROR  ]         multiqc : Oops! The 'slamdunk' MultiQC module broke...
                    Please copy the following traceback and report it at https://github.com/ewels/MultiQC/issues
                    (if possible, include a log file that triggers the error)
============================================================
Module slamdunk raised an exception: Traceback (most recent call last):
  File "/groups/zuber/zubarchive/USERS/tobias/dev/MultiQC/scripts/multiqc", line 346, in multiqc
    report.modules_output.append(mod())
  File "/groups/zuber/zubarchive/USERS/tobias/dev/MultiQC/multiqc/modules/slamdunk/slamdunk.py", line 140, in __init__
    self.slamdunkOverallRatesPlot()
  File "/groups/zuber/zubarchive/USERS/tobias/dev/MultiQC/multiqc/modules/slamdunk/slamdunk.py", line 616, in slamdunkOverallRatesPlot
    plots.bargraph.plot([self.rates_data_plus,self.rates_data_minus], cats, pconfig)
  File "/groups/zuber/zubarchive/USERS/tobias/dev/MultiQC/multiqc/plots/bargraph.py", line 94, in plot
    thisdict = { 'name': cats[idx][c]['name'], 'data': thisdata }
KeyError: 'name'
============================================================